zulipterminal/model.py

```diff
@@ -516,6 +516,45 @@ def send_stream_message(self, stream: str, topic: str, content: str) -> bool:
             notify_if_message_sent_outside_narrow(composition, self.controller)
         return message_was_sent
 
+    def prevent_compose_box_in_stream(self, stream_identifier: Any) -> bool:
+        """
+        Identify whether a user is authorised to send message to the given
+        stream or not using stream_post_policy and/or is_announcement_policy
+        and allow/restrict the compose box from opening and accordingly
+        update the footer
+        """
+        if self.stream_dict[stream_identifier].get(
+            "stream_post_policy"
+        ) == 2 or self.stream_dict[stream_identifier].get("is_announcement_only"):
+            # checks whether the user is an admin/owner
+            if (
+                self.get_user_info(self.user_id).get("role") <= 200
+                or self.initial_data.get("is_admin", True)
+                or self.initial_data.get("is_owner", True)
+            ):
+                return None
+            else:
+                return "Only Admins and owners can type"
+        elif self.stream_dict[stream_identifier].get("stream_post_policy") == 4:
+            # checks whether the user is an admin/owner/moderator
+            if (
+                self.get_user_info(self.user_id).get("role") <= 300
+                or self.initial_data.get("is_admin", True)
+                or self.initial_data.get("is_owner", True)
+            ):
+                return None
+            else:
+                return "Only Admins, moderators and owners are allowd to type"
+        elif self.stream_dict[stream_identifier].get("stream_post_policy") == 3:
+            # checks whether the user is an admin/owner/moderator/full member
+            if self.get_user_info(self.user_id).get("role") <= 400:
```

Comment:
This includes all users; a "full user" is subtly different and can be time-limited (eg. so new members cannot write to some streams). Check the zulip help documentation.

Reply:
Yes, that is the definition which had been [suggested](https://chat.zulip.org/#narrow/stream/412-api-documentation/topic/stream_post_policy.20.28API.20docs.29/near/1319730) on the #api-documentation channel.
However, as suggested by Tim in the [#api-documentation>stream_post_policy](https://chat.zulip.org/#narrow/stream/412-api-documentation/topic/stream_post_policy.20.28API.20docs.29/near/1319836), I checked with the [help center documentation](https://zulip.com/help/roles-and-permissions) and here they have mentioned that a _full member_ is

> Some organization settings allow an organization to restrict the permissions of new members; Members who do not have those restrictions are called full members.

So it is not limited to the new members who are members for the `waiting_period_threshold`. Hence, I kept a separate conditional statement for this as well.

Reply:
Ah, OK, so you know about it but you've left the `waiting_period_threshold` part unimplemented at this point?

Reply:
I hadn't actually thought about its implementation yet. However, I'll look into it.
On a side note, I couldn't find any references to `waiting_period_threshold` in the terminal repository nor in the (API and Developer) documentation. Can you please guide me on that?

Reply:
I see you posted about that; you could check the server code in the meantime.